File: BookExamples/ViewModelDemo_SavedState/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.java
package com.ebookfrenzy.viewmodeldemo.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.SavedStateHandle;

public class MainViewModel extends ViewModel {

    private static final String RESULT_KEY = "Euro Value";
    private static final Float usd_to_eu_rate = 0.74F;
    private String dollarText = "";
    private SavedStateHandle savedStateHandle;
    private MutableLiveData<Float> result;

    public void setAmount(String value) {
        this.dollarText = value;
        Float convertedValue = Float.parseFloat(dollarText)*usd_to_eu_rate;
        result.setValue(convertedValue);
        savedStateHandle.set(RESULT_KEY, convertedValue);
    }

    public MainViewModel(SavedStateHandle savedStateHandle) {
        this.savedStateHandle = savedStateHandle;
        result = savedStateHandle.getLiveData(RESULT_KEY);
    }

    public MutableLiveData<Float> getResult()
    {
        return result;
    }
}

File: KotlinIguanaSamples/ViewModelDemo_SavedState/app/src/main/java/com/ebookfrenzy/viewmodeldemo/MainViewModel.kt
package com.ebookfrenzy.viewmodeldemo

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.SavedStateHandle

const val RESULT_KEY = "Euro Value"

class MainViewModel(private val savedStateHandle: SavedStateHandle) : ViewModel() {

    private val rate = 0.74f
    private var dollarText = ""
    private var result: MutableLiveData<Float> = savedStateHandle.getLiveData(RESULT_KEY)


    fun setAmount(value: String) {
        this.dollarText = value
        val convertedValue = value.toFloat() * rate
        result.value = convertedValue
        savedStateHandle[RESULT_KEY] = convertedValue
    }

    fun getResult(): MutableLiveData<Float> {
        return result
    }
}